### ISSUE TITLE
Add new FAIL_*_PROPERTY macros

### ISSUE THREAD

**Issue by tudortimi** [svunit/svunit#10]
When unit testing drivers, I've noticed that I need a lot of procedural code to describe my expectations on what the result of driving an item is. Procedural code is clunky for describing temporal behavior; SVAs are a much better fit.

One could use the 'expect' construct to "assert" that a certain property holds:

property some_property;
  // business code
endpropery

`SVTEST(some_test)
  // set up test

  expect (some_property)
  else
    `FAIL_IF(1)
`SVTEST_END

The 'expect(...)' stuff could be encapsulated in a `FAIL_UNLESS_PROPERTY(...) macro.

I'm not sure about support for 'expect' across EDA vendors. All it takes is a single one that doesn't support it and the code won't compile anymore.

I'll blog about this on my site and come up with a proof of concept.

**Comment by nosnhojn** [comment]
yes please. I think this would be a good idea. I know Srinivasan Venkataramanan is working on the svunit+sva now for dvcon india. he might be a good guy (at the right time) to team up with on something like this.

as an aside, think about doing this as a plugin. I'm not married to the existing asserts so if you come up with wildly different, simple and way better, I'm good with that.

**Comment by tudortimi** [comment]
Extra clarification: This isn't about verifying SVAs (what Srini is doing I think). It's about writing unit test expectations **as** SVAs

**Comment by nosnhojn** [comment]
yup. got it. just thought there may be potential for 2 good ideas to somehow come together. maybe/maybe not.
